src/screens/RegisterScreen.js

```diff
@@ -0,0 +1,198 @@
+import React, {useState} from 'react';
+import {
+  View,
+  Text,
+  StyleSheet,
+  Button,
+  TextInput,
+  TouchableOpacity,
+  Image,
+  SafeAreaView,
+} from 'react-native';
+import Icon from 'react-native-vector-icons/AntDesign';
+import Color from '../redux/constants/Color';
+import {useDispatch, useSelector} from 'react-redux';
+import {registerUser} from '../redux/action/Action';
+import TextInputComponent from '../components/TextInputComponent';
+
+const RegisterScreen = ({navigation}) => {
+  const [fullname, setFullName] = useState('');
+  const [email, setEmail] = useState('');
+  const [mobileno, setMobileNo] = useState('');
+  const [password, setPassword] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [fullnameError, setFullNameError] = useState('');
+  const [mobilenoError, setMobileNoError] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+
+  const dispatch = useDispatch();
+
+  const register = useSelector(state => state.user.register);
+
+  const complete = () => {
+    dispatch(
+      registerUser({
+        fullname: fullname,
+        email: email,
+        password: password,
+        mobileno: mobileno,
+      }),
+    );
+    //navigation.navigate('HomeScreen');
+  };
+
+  const fullnameValidator = () => {
+    if (fullname === '') {
+      setFullNameError('fullname cannot be empty');
+    } else {
+      setFullNameError('');
+    }
+  };
+
+  const emailValidator = () => {
+    if (email === '') {
+      setEmailError('email cannot be empty');
+    } else {
+      setEmailError('');
+    }
+  };
+
+  const mobilenoValidator = () => {
+    if (mobileno === '') {
+      setMobileNoError('mobile must be 10 characters long');
+    } else {
+      setMobileNoError('');
+    }
+  };
+
+  const passwordValidator = () => {
+    if (password === '') {
+      setPasswordError('password must be 8 characters long');
+    } else {
+      setPasswordError('');
+    }
+  };
```

Comment:
add more validations and try to make these in less code and less repeatation

Reply:
complete it using regex validation